numpy/polynomial/legendre.py

```diff
@@ -655,15 +657,8 @@ def legpow(c, pow, maxpower=16):
         raise ValueError("Power is too large")
     elif power == 0:
         return np.array([1], dtype=c.dtype)
-    elif power == 1:
-        return c
     else:
-        # This can be made more efficient by using powers of two
-        # in the usual way.
-        prd = c
-        for i in range(2, power + 1):
-            prd = legmul(prd, c)
-        return prd
+        return reduce(legmul, [c]*power)
```

Comment:
These ones look fine, but the power of two comment should stay if you don't want to address it yet.

Reply:
Roger Wilco